cmd/osbuild-jobsite-builder/main.go

```diff
@@ -119,120 +118,94 @@ func (builder *Builder) GetState() State {
-	builder.GuardState(StateProvision)
-
-	if r.Method != "PUT" {
-		return fmt.Errorf("Builder.HandleProvision: Unexpected request method")
-	}
-
+func (b *Builder) HandleProvision(w http.ResponseWriter, r *http.Request) error {
 	logrus.WithFields(logrus.Fields{"argBuildPath": argBuildPath}).Debug("Builder.HandleProvision: Opening manifest.json")
 
-	dst, err := os.OpenFile(
+	dst, err := os.Create(
 		path.Join(argBuildPath, "manifest.json"),
-		os.O_WRONLY|os.O_CREATE|os.O_EXCL,
-		0400,
 	)
 
-	defer func() {
-		if cerr := dst.Close(); cerr != nil {
-			err = cerr
-		}
-	}()
-
 	if err != nil {
-		return fmt.Errorf("Builder.HandleProvision: Failed to open manifest.json")
+		w.WriteHeader(http.StatusInternalServerError)
+		return fmt.Errorf("Builder.HandleProvision: Failed to open manifest.json: %v", err)
 	}
 
 	logrus.Debug("Builder.HandleProvision: Writing manifest.json")
 
 	_, err = io.Copy(dst, r.Body)
 
 	if err != nil {
-		return fmt.Errorf("Builder.HandleProvision: Failed to write manifest.json")
+		w.WriteHeader(http.StatusInternalServerError)
+		return fmt.Errorf("Builder.HandleProvision: Failed to write manifest.json: %v", err)
 	}
 
-	w.WriteHeader(http.StatusCreated)
-
-	if _, err := w.Write([]byte(`done`)); err != nil {
-		return fmt.Errorf("Builder.HandleProvision: Failed to write response")
+	if err = dst.Close(); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return fmt.Errorf("Builder.HandleProvision: Failed to close manifest.json: %v", err)
 	}
 
 	logrus.Info("Builder.HandleProvision: Done")
 
-	builder.SetState(StatePopulate)
+	w.WriteHeader(http.StatusCreated)
+	b.SetState(StatePopulate)
 
 	return nil
 }
 
-func (builder *Builder) HandlePopulate(w http.ResponseWriter, r *http.Request) error {
-	builder.GuardState(StatePopulate)
-
-	if r.Method != "POST" {
-		return fmt.Errorf("Builder.HandlePopulate: unexpected request method")
-	}
-
-	w.WriteHeader(http.StatusOK)
-
-	if _, err := w.Write([]byte(`done`)); err != nil {
-		return fmt.Errorf("Builder.HandlePopulate: Failed to write response")
-	}
-
+func (b *Builder) HandlePopulate(w http.ResponseWriter, r *http.Request) error {
 	logrus.Info("Builder.HandlePopulate: Done")
 
-	builder.SetState(StateBuild)
+	w.WriteHeader(http.StatusOK)
+	b.SetState(StateBuild)
 
 	return nil
 }
 
-func (builder *Builder) HandleBuild(w http.ResponseWriter, r *http.Request) error {
-	builder.GuardState(StateBuild)
-
-	if r.Method != "POST" {
-		return fmt.Errorf("Builder.HandleBuild: Unexpected request method")
-	}
-
+func (b *Builder) HandleBuild(w http.ResponseWriter, r *http.Request) error {
 	var buildRequest BuildRequest
 
-	var err error
-
-	if err = json.NewDecoder(r.Body).Decode(&buildRequest); err != nil {
+	if err := json.NewDecoder(r.Body).Decode(&buildRequest); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
 		return fmt.Errorf("HandleBuild: Failed to decode body")
```

Comment:
It would be nice to include the origin error here, e.g. `fmt.Errorf("HandleBuild: Failed to decode body: %w", err)`